sources/models/src/ephemeral_storage.rs

```diff
@@ -18,12 +18,58 @@ impl Display for Filesystem {
     }
 }
 
+/// Storage type preferences for ephemeral storage
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Preference {
+    pub ephemeral_disk: bool,
+    pub ebs_volume: bool,
+}
+impl Display for Preference {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut first = false;
+        let mut write = |s| {
+            if first {
+                first = false;
+            } else {
+                f.write_str("+")?;
+            }
+            f.write_str(s)
+        };
+        if self.ephemeral_disk {
+            write("ephemeral-disk")?;
+        }
+        if self.ebs_volume {
+            write("ebs-volume")?;
+        }
+        Ok(())
+    }
+}
+impl<'a> TryFrom<&'a str> for Preference {
+    type Error = &'a str;
+
+    fn try_from(value: &'a str) -> Result<Self, Self::Error> {
+        let mut preference = Self {
+            ephemeral_disk: false,
+            ebs_volume: false,
+        };
+        for i in value.split("+") {
+            match i {
+                "ephemeral-disk" => preference.ephemeral_disk = true,
+                "ebs-volume" => preference.ebs_volume = true,
+                x => return Err(x),
+            }
```

Comment:
Should `""` be allowed here? It would allow defining an empty set (not very useful) and superfluous `+`s, e.g. `ephemeral-disk+`. The main benefit would probably be enabling round-tripping through `Display` and `TryFrom<&str>` for the empty set case.